CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,77 @@
+our community a harassment-free experience for everyone, regardless of age, body
+size, disability, ethnicity, sex characteristics, gender identity and expression,
+level of experience, education, socio-economic status, nationality, personal
+appearance, race, religion, or sexual identity and orientation.
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment
+include:
+
+* Using welcoming and inclusive language
+* Being respectful of differing viewpoints and experiences
+* Gracefully accepting constructive criticism
+* Focusing on what is best for the community
+* Showing empathy towards other community members
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery and unwelcome sexual attention or
+  advances
+* Trolling, insulting/derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or electronic
+  address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Our Responsibilities
+
+Project maintainers are responsible for clarifying the standards of acceptable
+behavior and are expected to take appropriate and fair corrective action in
+response to any instances of unacceptable behavior.
+
+Project maintainers have the right and responsibility to remove, edit, or
+reject comments, commits, code, wiki edits, issues, and other contributions
+that are not aligned to this Code of Conduct, or to ban temporarily or
+permanently any contributor for other behaviors that they deem inappropriate,
+threatening, offensive, or harmful.
+
+## Scope
+
+This Code of Conduct applies within all project spaces, and it also applies when
+an individual is representing the project or its community in public spaces.
+Examples of representing a project or community include using an official
+project e-mail address, posting via an official social media account, or acting
+as an appointed representative at an online or offline event. Representation of
+a project may be further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at <pcapplusplus@gmail.com>. All
+complaints will be reviewed and investigated and will result in a response that
+is deemed necessary and appropriate to the circumstances. The project team is
+obligated to maintain confidentiality with regard to the reporter of an incident.
+Further details of specific enforcement policies may be posted separately.
+
+Project maintainers who do not follow or enforce the Code of Conduct in good
+faith may face temporary or permanent repercussions as determined by other
+members of the project's leadership.
+
+## Attribution
+
+This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 1.4,
```

Comment:
- I can see the latest version of this CoC is 2.1. What's the difference between the versions?
- I admit I know nothing about CoC. Why did you choose Contributor Covenant? What are the other options for adopting CoC?

Reply:
There is no huge difference. I think CoC 1.4 uses simpler terms than 2.1 since 2.1 is more formal than 1.4. For non-english native speakers, it can be easier to understand terms like `Project` instead of `Community` which can be easily referred as "PcapPlusPlus project"

> Project maintainers are responsible for clarifying the standards of acceptable behavior ...

instead of

> Community leaders are responsible for clarifying and enforcing our standards of
acceptable behavior ...

or

> This Code of Conduct applies within all project spaces,

instead of

> This Code of Conduct applies within all community spaces

But if we decide to use the latest version, updating CoC will be easier if needed.


About why Contributor Covenant I researched the web a bit. When you search about a CoC most of time found documents/examples are actually a company related. But Contributor Covenant is directly created for Open-Source Communities. So, I found it more suitable for PcapPlusPlus.

Reply:
A side note we can add PcapPlusPlus to adopters in contributor covenant with a PR which mentioned [here](https://www.contributor-covenant.org/adopters/) 🙂

Reply:
I just bumped into [this GitHub guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project) on adding a CoC, and their template also points to Contributor Covenant 🙂 
![image](https://github.com/seladb/PcapPlusPlus/assets/9059541/a0666e7a-d688-4283-b7ff-f2b217127426)